checks/mysql/check.go

```diff
@@ -0,0 +1,46 @@
+package mysql
+
+import (
+	"database/sql"
+)
+
+// Config is the MySQL checker configuration settings container.
+type Config struct {
+	// DSN is the MySQL instance connection DSN. Required.
+	DSN string
+
+	// LogFunc is the callback function for errors logging during check.
+	// If not set logging is skipped.
+	LogFunc func(err error, details string, extra ...interface{})
+}
+
+// New creates new MySQL health check that verifies the following:
+// - connection establishing
+// - doing the ping command
+func New(config Config) func() error {
+	if config.LogFunc == nil {
+		config.LogFunc = func(err error, details string, extra ...interface{}) {}
+	}
+
+	return func() error {
+		db, err := sql.Open("mysql", config.DSN)
+		if err != nil {
+			config.LogFunc(err, "MySQL health check failed during connect")
+			return err
+		}
+
+		defer func() {
+			if err = db.Close(); err != nil {
+				config.LogFunc(err, "MySQL health check failed during connection closing")
+			}
+		}()
+
+		err = db.Ping()
+		if err != nil {
+			config.LogFunc(err, "MySQL health check failed during ping")
+			return err
+		}
+
+		return nil
+	}
```

Comment:
Connection handling in `Go` is different from `PHP`. Since `go` is a long-running process the driver knows how to keep and maintain a connection pool. In this respect there are few considerations:

1. There is no need to `Close()` the connection
2. `sql.Open()` will open a lazy connection, i.e. will not connect for real before `Ping()` is called
3. since the driver knows how to maintain the pool - the given implementation will check if it is possible to connect to db vs pinging the connection that the application is using.

- [sql.Open](https://golang.org/src/database/sql/sql.go?s=19074:19131#L647) - read the comment
- [connection pool](http://go-database-sql.org/connection-pool.html)

Reply:
Thank you for your comments! That makes sense:) It was a nice read :+1:  I removed however the Ping() and made it consistent with the postgres healthcheck that does inserts, selects and deletes as well.